lib/util.ml

```diff
@@ -391,18 +409,47 @@ module Webhook = struct
     | Some state ->
       let* build_failed_steps = get_failed_steps () in
 
-      (* return only the new failed steps. Keep all the failed steps in state, but keep the
-         original failed steps urls for the ones that intersect, instead of the new ones. *)
       let steps_intersect = FailedStepSet.inter state.steps build_failed_steps in
       let new_failed_steps = FailedStepSet.diff build_failed_steps state.steps in
-      let updated_steps = FailedStepSet.union steps_intersect new_failed_steps in
-      Stringtbl.replace repo_state.failed_steps repo_key { steps = updated_steps; last_build = n.build.number };
+
+      (* escalate steps that have been broken for longer than the threshold. Escalate again after every hour *)
+      let steps_to_escalate =
+        match get_escalation_threshold cfg n with
+        | None -> new_failed_steps
+        | Some escalation_threshold ->
+          FailedStepSet.filter
+            (fun step ->
+              match step.escalated_at with
+              | None ->
+                (* escalate steps that have been broken for longer than the threshold *)
+                is_past_span step.created_at escalation_threshold
+              | Some escalated_at ->
+                (* escalate again if the escalated_at time is more than 1 hour ago *)
+                let one_hour = Ptime.Span.of_int_s (60 * 60) in
+                is_past_span escalated_at one_hour)
```

Comment:
Should this constant be part of the config instead? Like `reescalate_notification_delay` or something like that? It seems quite specific to some use-case (e.g., some teams might work on more like a day schedule).

Reply:
yeah, maybe it should, but i want to avoid flooding the config with options. 

I think this is a sensible default: you have a first escalation warning after a certain period of time that you consider a bit too long for the build to have been broken and unattended, and then every extra hour you notify again when you have new builds that are broken due to the original author not fixing the issues.

In any case it's better to test first and tweak afterwards. You know the old "how you use your product/how users use the product" meme.. 🤷‍♂️